components/common/HierarchicalCheckbox.tsx

```diff
@@ -0,0 +1,213 @@
+// Copyright 2022 Curtin University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Alex Massen-Hane
+
+import { Field, FastField, useField, useFormikContext } from "formik";
+import { ChevronRightIcon, ChevronDownIcon } from "@chakra-ui/icons";
+import React, { memo, useEffect, useCallback, ReactElement } from "react";
+import { QueryForm } from "../filter/FilterForm";
+import {
+  Checkbox,
+  Accordion,
+  AccordionButton,
+  AccordionItem,
+  AccordionPanel,
+  Text,
+  useAccordionItemState,
+  Flex,
+} from "@chakra-ui/react";
+
+export interface HierarchicalCheckboxProps {
+  treeObject: RecursiveTreeCheckboxProps;
+}
+
+const HierarchicalCheckbox = ({ treeObject }: HierarchicalCheckboxProps): ReactElement => {
+  return (
+    <Accordion allowMultiple variant="hierarchicalCheckbox" reduceMotion={true}>
+      {RecursiveTreeCheckbox(treeObject)}
+    </Accordion>
+  );
+};
+
+export interface RecursiveTreeCheckboxProps {
+  level: number;
+  type: string;
+  text: string;
+  children?: Array<RecursiveTreeCheckboxProps>;
+}
+
+// Recursively go down branches and render checkboxes and input values into Formik
+const RecursiveTreeCheckbox = (branch: RecursiveTreeCheckboxProps): ReactElement => {
+  if (branch.children === undefined) {
+    return <></>;
+  } else {
```

Comment:
With the above suggestions regarding recursion / theming, we could simplify the RecursiveTreeCheckboxProps type,  e.g. level isn't needed anymore. We won't have to add the if(branch.children) checks with these type definitions too.

Example types:
```typescript
export interface HierarchicalCheckboxParent {
  type: string;
  text: string;
  children: Array<HierarchicalCheckboxChild>;
}

export interface HierarchicalCheckboxChild {
  type: string;
  text: string;
}
```

Example config:
```typescript
export const regionsConfig: Array<HierarchicalCheckboxParent> = [
  {
    type: "region",
    text: "Africa",
    children: [
      {
        type: "subregion",
        text: "Northern Africa",
      },
      {
        type: "subregion",
        text: "Sub-Saharan Africa",
      },
    ],
  },
  {
    type: "region",
    text: "Americas",
    children: [
      {
        type: "subregion",
        text: "Latin America and the Caribbean",
      },
      {
        type: "subregion",
        text: "Northern America",
      },
    ],
  },
];
```